backend/routes/follows.js

```diff
@@ -4,56 +4,113 @@ import { Stock } from "../models/stock.js";
 
 const router = express.Router();
 
+const checkSession = (req, res, next) => {
+    const user = req.session.user;
+    if (!user) {
+        res.status(401).json({ error: "Missing Session." });
+    } else {
```

Comment:
Super nit: There is nothing wrong with IF - ELSE, but usually it "looks" better to return early and avoid the use of the ELSE clause. Like you did below in the "/follows" route.